internal/pkg/agent/control/v2/server/server.go

```diff
@@ -189,19 +187,24 @@ func (s *Server) Upgrade(ctx context.Context, request *cproto.UpgradeRequest) (*
 
 // DiagnosticAgent returns diagnostic information for this running Elastic Agent.
 func (s *Server) DiagnosticAgent(ctx context.Context, _ *cproto.DiagnosticAgentRequest) (*cproto.DiagnosticAgentResponse, error) {
-	res := make([]*cproto.DiagnosticFileResult, 0, len(s.diagHooks))
-	for _, h := range s.diagHooks {
+	diagHooks := make([]diagnostics.Hook, 0)
+	for _, fn := range s.diagHooksFn {
```

Comment:
We want the log collection hooks to like [`coordinator.addLogHooks`](https://github.com/elastic/elastic-agent/pull/1703/files#diff-efef2d2536ab052feff95a0b0565325806b28e733bb9f2abc319b646194d4de5R545) to be called when the diagnostics are collected so that all files have a hook

Reply:
i'll roll it back